src/Apps/CollectorProfile/Routes/Saves2/CollectorProfileSaves2Route.tsx

```diff
@@ -39,8 +48,30 @@ const CollectorProfileSaves2Route: FC<CollectorProfileSaves2RouteProps> = ({
   // Always display "Saved Artwork" collection first in the list
   const savedCollections = [savedCollection, ...otherCollections]
 
+  const onHistoryChanged = useCallback(
+    (id: string | null) => {
+      setActiveCollectionId(id ?? savedCollectionId)
+    },
+    [savedCollectionId, setActiveCollectionId]
+  )
+
+  useSavesHistoryEntity({
+    onChanged: onHistoryChanged,
+  })
+
+  const onDeleteCollection = () => {
+    setActiveCollectionId(savedCollectionId)
+    window.history.replaceState(null, "", "/collector-profile/saves2")
+  }
```

Comment:
Note that just calling `history.pushState()` or `history.replaceState()` won't trigger a `popstate` event (and this code [accordingly](https://github.com/artsy/force/blob/0393e301e64d6e1559cd286455220f637456ad34/src/Apps/CollectorProfile/Routes/Saves2/Hooks/useSavesHistoryEntity.ts#L12-L27)). The `popstate` event will be triggered by doing a browser action such as a **click** on the back or forward button (or **calling** `history.back()` or `history.forward()` in JavaScript). 

For this reason, these changes were added using context